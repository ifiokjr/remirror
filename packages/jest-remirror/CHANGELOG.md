# jest-remirror

## 1.0.0-next.0

> 2020-07-05

### Major Changes

- The whole API for remirror has completely changed. These pre-release versions are a breaking
  change across all packages. The best way to know what's changed is to read the documentaion on the
  new documentation site `https://remirror.io`.
- 28bd8bea: This is a breaking change to the structure of published npm packages.

  - Move build directory from `lib` to `dist`
  - Remove option for multiple entry points. It is no longer possible to import module from
    '@remirror/core/lib/custom'
  - Only use one entry file.
  - Remove declaration source mapping for declaration files
  - Remove the src directory from being published.

- 7b817ac2: Rename all types and interfaces postfixed with `Params` to use the postfix `Parameter`.
  If your code was importing any matching interface you will need to update the name.
- dd16d45d: Rewrite packages using the new API

### Minor Changes

- 9a699e80: Upgrade dependencies to use v26.0.0 of jest.

### Patch Changes

- Updated dependencies [undefined]
- Updated dependencies [8334294e]
- Updated dependencies [28bd8bea]
- Updated dependencies [7b817ac2]
- Updated dependencies [9a699e80]
- Updated dependencies [dd16d45d]
- Updated dependencies [8334294e]
  - @remirror/pm@1.0.0-next.0
  - jest-prosemirror@1.0.0-next.0
  - remirror@1.0.0-next.0

## 0.13.1

### Patch Changes

- Updated dependencies [4dbb7461]
  - @remirror/core-extensions@0.13.1
  - @remirror/react@0.13.1

## 0.11.1

### Patch Changes

- 000fdfb0: Upgraded external dependencies with major releases.
- Updated dependencies [d2a288aa]
- Updated dependencies [5888a7aa]
- Updated dependencies [000fdfb0]
  - jest-prosemirror@0.8.2
  - @remirror/core-extensions@0.11.1

## 0.11.0

### Patch Changes

- Updated dependencies [026d4238]
- Updated dependencies [69d00c62]
- Updated dependencies [c2237aa0]
  - @remirror/react@0.11.0
  - @remirror/core@0.11.0
  - @remirror/core-extensions@0.11.0

## 0.8.1

### Patch Changes

- Updated dependencies [0300d01c]
  - @remirror/core@0.9.0
  - @remirror/core-extensions@0.7.6
  - jest-prosemirror@0.8.1
  - @remirror/react-utils@0.7.6
  - @remirror/react@0.7.7

## 0.8.0

### Minor Changes

- 527395be: `renderEditor` now accepts PrioritizedExtensions for more flexible testing.

### Patch Changes

- Updated dependencies [24f83413]
- Updated dependencies [24f83413]
- Updated dependencies [24f83413]
  - @remirror/core@0.8.0
  - jest-prosemirror@0.8.0
  - @remirror/core-extensions@0.7.5
  - @remirror/react-utils@0.7.5
  - @remirror/react@0.7.6

## 0.7.4

### Patch Changes

- 7380e18f: Update repository url from ifiokjr/remirror to remirror/remirror to reflect new GitHub
  organisation.
- Updated dependencies [10419145]
- Updated dependencies [7380e18f]
  - @remirror/core-extensions@0.7.4
  - @remirror/core@0.7.4
  - @remirror/react-utils@0.7.4
  - @remirror/react@0.7.5
  - jest-prosemirror@0.7.4

## 0.7.3

### Patch Changes

- 5f85c0de: Bump a new version to test out the changeset API.
- Updated dependencies [5f85c0de]
  - @remirror/core@0.7.3
  - @remirror/core-extensions@0.7.3
  - @remirror/react@0.7.3
  - @remirror/react-utils@0.7.3
  - jest-prosemirror@0.7.3
