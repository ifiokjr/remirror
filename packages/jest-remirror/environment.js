require('./lib/setup-environment').setupRemirrorEnvironment();
