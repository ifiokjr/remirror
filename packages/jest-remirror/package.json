{
  "name": "jest-remirror",
  "version": "1.0.0-next.51",
  "description": "A text editor for react built with prosemirror",
  "keywords": [
    "prosemirror",
    "react",
    "ssr",
    "text editor"
  ],
  "homepage": "https://github.com/remirror/remirror/tree/HEAD/packages/jest-remirror",
  "repository": "https://github.com/remirror/remirror/tree/HEAD/packages/jest-remirror",
  "license": "MIT",
  "contributors": [
    "Ifiok Jr. <ifiokotung@gmail.com>"
  ],
  "main": "dist/jest-remirror.cjs.js",
  "types": "dist/jest-remirror.cjs.d.ts",
  "files": [
    "dist",
    "environment"
  ],
  "dependencies": {
    "@babel/runtime": "^7.12.0",
    "@remirror/core": "1.0.0-next.51",
    "@remirror/dom": "1.0.0-next.51",
    "@remirror/pm": "1.0.0-next.51",
    "@remirror/preset-core": "1.0.0-next.51",
    "@testing-library/dom": "^7.26.3",
    "@types/sanitize-html": "^1.27.0",
    "jest-prosemirror": "1.0.0-next.51",
    "jest-snapshot": "^26.6.1",
    "mutationobserver-shim": "^0.3.7",
    "sanitize-html": "^2.1.1"
  },
  "devDependencies": {
    "@types/node": "^14.14.3",
    "jest": "^26.6.1"
  },
  "peerDependencies": {
    "@types/node": "*",
    "jest": ">= 24"
  },
  "peerDependenciesMeta": {
    "@types/node": {
      "optional": true
    }
  },
  "publishConfig": {
    "access": "public"
  },
  "preconstruct": {
    "entrypoints": [
      ".",
      "environment"
    ]
  }
}
