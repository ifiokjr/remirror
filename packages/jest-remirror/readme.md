# jest-remirror

<div align="center">
	<br />
	<div>
		<img width="300" src="../../support/assets/logo-icon.svg" alt="remirror" />
    <h1 align="center">jest-remirror</h1>
	</div>
    <br />
    <br />
    <br />
    <br />
</div>

[![LICENSE](https://img.shields.io/npm/l/jest-remirror.svg?style=for-the-badge)](https://github.com/ifiokjr/remirror/blob/master/LICENSE) [![npm](https://img.shields.io/npm/dm/jest-remirror.svg?style=for-the-badge&logo=npm)](https://www.npmjs.com/package/jest-remirror) [![Dependencies (path)](https://img.shields.io/david/ifiokjr/remirror.svg?logo=npm&path=@remirror%2Fextension-mention&style=for-the-badge)](https://github.com/ifiokjr/remirror/blob/master/jest-remirror/package.json) [![NPM](https://img.shields.io/npm/l/jest-remirror.svg?style=for-the-badge)](https://github.com/ifiokjr/remirror/blob/master/LICENSE) [![GitHub issues by-label](https://img.shields.io/github/issues/ifiokjr/remirror/package%3A%20jest-remirror.svg?label=Open%20Issues&logo=github&style=for-the-badge)](https://github.com/ifiokjr/remirror/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3Apackage%3A%20jest-remirror) [![GitHub pull requests by-label](https://img.shields.io/github/issues-pr/ifiokjr/remirror/package%3A%20jest-remirror.svg?label=Open%20Pull%20Requests&logo=github&style=for-the-badge)](https://github.com/ifiokjr/remirror/pulls?utf8=%E2%9C%93&q=is%3Apr+is%3Aopen+sort%3Aupdated-desc+label%3Apackage%3A%20jest-remirror)

## Installation

```bash
yarn add jest-remirror
```

## Getting started

API and docs still being worked on.

## Acknowledgements

This package borrows very heavily from [@atlaskit/editor-test-helpers](https://www.npmjs.com/package/@atlaskit/editor-test-helpers)
