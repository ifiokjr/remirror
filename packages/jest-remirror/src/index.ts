export * from './builder';
export * from './transactions';
export * from './matchers';
export * from './jsdom-polyfills';
export * from './test-schema';
export * from './test-editor';
export * from './setup-environment';

declare global {
  interface Window {
    hasWarnedAboutCancelAnimationFramePolyfill?: boolean;
    hasWarnedAboutJsdomFixtures: boolean;
    ignoreAllJSDOMWarnings: boolean;
  }
}
