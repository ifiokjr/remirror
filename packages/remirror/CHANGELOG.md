# remirror

## 0.7.6

### Patch Changes

- Updated dependencies [0300d01c]
  - @remirror/core@0.9.0
  - @remirror/core-extensions@0.7.6
  - @remirror/react@0.7.7

## 0.7.5

### Patch Changes

- Updated dependencies [24f83413]
- Updated dependencies [24f83413]
  - @remirror/core@0.8.0
  - @remirror/core-extensions@0.7.5
  - @remirror/react@0.7.6

## 0.7.4

### Patch Changes

- 7380e18f: Update repository url from ifiokjr/remirror to remirror/remirror to reflect new GitHub
  organisation.
- Updated dependencies [10419145]
- Updated dependencies [7380e18f]
  - @remirror/core-extensions@0.7.4
  - @remirror/core@0.7.4
  - @remirror/react@0.7.5

## 0.7.3

### Patch Changes

- 5f85c0de: Bump a new version to test out the changeset API.
- Updated dependencies [5f85c0de]
  - @remirror/core@0.7.3
  - @remirror/core-extensions@0.7.3
  - @remirror/react@0.7.3
