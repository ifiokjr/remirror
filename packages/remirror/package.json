{
  "name": "remirror",
  "version": "1.0.0-next.4",
  "description": "One editing package to rule them all, one editing package to find them.",
  "keywords": [
    "prosemirror",
    "react",
    "remirror",
    "wysiwyg",
    "editing",
    "ssr",
    "text editor"
  ],
  "homepage": "https://github.com/remirror/remirror/tree/HEAD/packages/remirror",
  "repository": "https://github.com/remirror/remirror/tree/HEAD/packages/remirror",
  "license": "MIT",
  "contributors": [
    "Ifiok Jr. <ifiokotung@gmail.com>"
  ],
  "sideEffects": false,
  "main": "dist/remirror.cjs.js",
  "module": "dist/remirror.esm.js",
  "types": "dist/remirror.cjs.d.ts",
  "files": [
    "core",
    "dist",
    "dom",
    "theme",
    "extension",
    "preset",
    "react",
    "styles"
  ],
  "dependencies": {
    "@babel/runtime": "^7.10.4",
    "@remirror/core": "^1.0.0-next.4",
    "@remirror/core-constants": "^1.0.0-next.4",
    "@remirror/core-helpers": "^1.0.0-next.4",
    "@remirror/core-types": "^1.0.0-next.4",
    "@remirror/core-utils": "^1.0.0-next.4",
    "@remirror/dom": "^1.0.0-next.4",
    "@remirror/extension-auto-link": "^1.0.0-next.4",
    "@remirror/extension-base-keymap": "^1.0.0-next.4",
    "@remirror/extension-bidi": "^1.0.0-next.4",
    "@remirror/extension-blockquote": "^1.0.0-next.4",
    "@remirror/extension-bold": "^1.0.0-next.4",
    "@remirror/extension-code": "^1.0.0-next.4",
    "@remirror/extension-code-block": "^1.0.0-next.4",
    "@remirror/extension-collaboration": "^1.0.0-next.4",
    "@remirror/extension-diff": "^1.0.0-next.4",
    "@remirror/extension-doc": "^1.0.0-next.4",
    "@remirror/extension-drop-cursor": "^1.0.0-next.4",
    "@remirror/extension-emoji": "^1.0.0-next.4",
    "@remirror/extension-epic-mode": "^1.0.0-next.4",
    "@remirror/extension-gap-cursor": "^1.0.0-next.4",
    "@remirror/extension-hard-break": "^1.0.0-next.4",
    "@remirror/extension-heading": "^1.0.0-next.4",
    "@remirror/extension-history": "^1.0.0-next.4",
    "@remirror/extension-horizontal-rule": "^1.0.0-next.4",
    "@remirror/extension-image": "^1.0.0-next.4",
    "@remirror/extension-italic": "^1.0.0-next.4",
    "@remirror/extension-link": "^1.0.0-next.4",
    "@remirror/extension-mention": "^1.0.0-next.4",
    "@remirror/extension-paragraph": "^1.0.0-next.4",
    "@remirror/extension-placeholder": "^1.0.0-next.4",
    "@remirror/extension-position-tracker": "^1.0.0-next.4",
    "@remirror/extension-positioner": "^1.0.0-next.4",
    "@remirror/extension-react-component": "^1.0.0-next.4",
    "@remirror/extension-react-ssr": "^1.0.0-next.4",
    "@remirror/extension-search": "^1.0.0-next.4",
    "@remirror/extension-strike": "^1.0.0-next.4",
    "@remirror/extension-text": "^1.0.0-next.4",
    "@remirror/extension-trailing-node": "^1.0.0-next.4",
    "@remirror/extension-underline": "^1.0.0-next.4",
    "@remirror/extension-yjs": "^1.0.0-next.4",
    "@remirror/preset-core": "^1.0.0-next.4",
    "@remirror/preset-embed": "^1.0.0-next.4",
    "@remirror/preset-list": "^1.0.0-next.4",
    "@remirror/preset-react": "^1.0.0-next.4",
    "@remirror/preset-social": "^1.0.0-next.4",
    "@remirror/preset-table": "^1.0.0-next.4",
    "@remirror/preset-wysiwyg": "^1.0.0-next.4",
    "@remirror/react": "^1.0.0-next.4",
    "@remirror/react-social": "^1.0.0-next.4",
    "@remirror/react-utils": "^1.0.0-next.4",
    "@remirror/react-wysiwyg": "^1.0.0-next.4",
    "@remirror/theme": "^1.0.0-next.4"
  },
  "devDependencies": {
    "@remirror/pm": "1.0.0-next.4",
    "@types/react": "^16.9.43",
    "@types/react-dom": "^16.9.8",
    "react": "^16.13.1",
    "react-dom": "^16.13.1"
  },
  "peerDependencies": {
    "@remirror/pm": "^1.0.0-next.4",
    "@types/react": "^16.8.0",
    "@types/react-dom": "^16.8.0",
    "react": "^16.8.0",
    "react-dom": "^16.8.0"
  },
  "peerDependenciesMeta": {
    "@types/react": {
      "optional": true
    },
    "@types/react-dom": {
      "optional": true
    },
    "react": {
      "optional": true
    },
    "react-dom": {
      "optional": true
    }
  },
  "publishConfig": {
    "access": "public"
  },
  "preconstruct": {
    "entrypoints": [
      ".",
      "core",
      "core/constants",
      "core/helpers",
      "core/types",
      "core/utils",
      "dom",
      "extension/auto-link",
      "extension/base-keymap",
      "extension/blockquote",
      "extension/bold",
      "extension/bidi",
      "extension/code-block",
      "extension/code",
      "extension/collaboration",
      "extension/doc",
      "extension/drop-cursor",
      "extension/emoji",
      "extension/epic-mode",
      "extension/gap-cursor",
      "extension/hard-break",
      "extension/heading",
      "extension/history",
      "extension/horizontal-rule",
      "extension/image",
      "extension/italic",
      "extension/link",
      "extension/mention",
      "extension/paragraph",
      "extension/placeholder",
      "extension/position-tracker",
      "extension/positioner",
      "extension/react-component",
      "extension/react-ssr",
      "extension/search",
      "extension/strike",
      "extension/text",
      "extension/diff",
      "extension/trailing-node",
      "extension/underline",
      "extension/yjs",
      "preset/core",
      "preset/embed",
      "preset/list",
      "preset/react",
      "preset/social",
      "preset/table",
      "preset/wysiwyg",
      "react",
      "react/social",
      "react/wysiwyg",
      "react/utils",
      "theme"
    ]
  }
}
