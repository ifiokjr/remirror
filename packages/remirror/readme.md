<div align="center">
	<br />
	<div align="center">
		<img width="300" src="https://cdn.jsdelivr.net/gh/ifiokjr/remirror/support/assets/logo-icon.svg" alt="remirror" />
    <h1 align="center">remirror</h1>
	</div>
    <br />
    <br />
    <br />
    <br />
</div>

[![npm bundle size (scoped)](https://img.shields.io/bundlephobia/minzip/remirror.svg?style=for-the-badge)](https://bundlephobia.com/result?p=remirror) [![npm](https://img.shields.io/npm/dm/remirror.svg?style=for-the-badge&logo=npm)](https://www.npmjs.com/package/remirror) [![Dependencies (path)](https://img.shields.io/david/ifiokjr/remirror.svg?logo=npm&path=packages%2Fremirror&style=for-the-badge)](https://github.com/ifiokjr/remirror/blob/master/remirror/package.json) [![NPM](https://img.shields.io/npm/l/remirror.svg?style=for-the-badge)](https://github.com/ifiokjr/remirror/blob/master/LICENSE) [![GitHub issues by-label](https://img.shields.io/github/issues/ifiokjr/remirror/package%3A%remirror.svg?label=Open%20Issues&logo=github&style=for-the-badge)](https://github.com/ifiokjr/remirror/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3Apackage%3A%remirror) [![GitHub pull requests by-label](https://img.shields.io/github/issues-pr/ifiokjr/remirror/package%3A%remirror.svg?label=Open%20Pull%20Requests&logo=github&style=for-the-badge)](https://github.com/ifiokjr/remirror/pulls?utf8=%E2%9C%93&q=is%3Apr+is%3Aopen+sort%3Aupdated-desc+label%3Apackage%3A%remirror)

Remirror is an extensible text-editor for react, built on top of Prosemirror. It aims to be **the** goto editor for a reliable editing experience across all JavaScript and user-facing environments.

The project is still in its early days and several of the ideas featured here still need to be fleshed out.

## Getting Started

### Prerequisites

- Typescript `>= 3.3` - Plans to use the latest `as const` syntax
- React `>= 16.8` - This project relies on hooks
- Yarn `>= 1.13` - It may work with previous versions as well

## Installation

```bash
yarn add remirror
```

The following is a small example which renders a floating menu and enables the extensions `Bold`, `Italic` and `Underline`.

```ts
import React, { FC, FunctionComponent, MouseEventHandler, useState } from 'react';

import {
  EMPTY_OBJECT_NODE,
  Bold,
  Italic,
  Underline,
  bubblePositioner,
  ManagedRemirrorEditor,
  RemirrorEventListener,
  RemirrorExtension,
  RemirrorManager,
  RemirrorProps,
  useRemirrorContext,
} from 'remirror';

const runAction = (action: () => void): MouseEventHandler<HTMLElement> => e => {
  e.preventDefault();
  action();
};

const SimpleFloatingMenu: FC = () => {
  const { getPositionerProps, actions } = useRemirrorContext(); // Pull in injected props from context

  const props = getPositionerProps({
    positionerId: 'bubble',
    ...bubblePositioner,
  });
  return (
    <div
      style={{
        position: 'absolute',
        bottom: props.isActive ? props.bottom : -9999,
        left: props.isActive ? props.left : -9999,
      }}
      ref={props.ref}
    >
      <button
        style={{
          backgroundColor: actions.bold.isActive() ? 'white' : 'pink',
          fontWeight: actions.bold.isActive() ? 600 : 300,
        }}
        disabled={!actions.bold.isEnabled()}
        onClick={runAction(actions.bold.command)}
      >
        b
      </button>
      <button
        style={{
          backgroundColor: actions.italic.isActive() ? 'white' : 'pink',
          fontWeight: actions.italic.isActive() ? 600 : 300,
        }}
        disabled={!actions.italic.isEnabled()}
        onClick={runAction(actions.italic.command)}
      >
        i
      </button>
      <button
        style={{
          backgroundColor: actions.underline.isActive() ? 'white' : 'pink',
          fontWeight: actions.underline.isActive() ? 600 : 300,
        }}
        disabled={!actions.underline.isEnabled()}
        onClick={runAction(actions.underline.command)}
      >
        u
      </button>
    </div>
  );
};

const EditorLayout: FunctionComponent = () => {
  return (
    <RemirrorManager>
      <RemirrorExtension Constructor={Bold} />
      <RemirrorExtension Constructor={Italic} />
      <RemirrorExtension Constructor={Underline} />
      <ManagedRemirrorEditor
        attributes={{ 'data-test-id': 'editor-instance' }}
        onChange={onChange}
        placeholder='Start typing for magic...'
        autoFocus={true}
        initialContent={EMPTY_OBJECT_NODE}
      >
        <SimpleFloatingMenu />
      </ManagedRemirrorEditor>
    </RemirrorManager>
  );
};
```

The above example uses hooks but you can just as easily rely on Higher Order Components (HOC's) to wrap your component.

In a similar fashion Higher Order Components (HOC's) can be used to wrap a component.

```ts
import { withRemirror } from 'remirror';

// ...

function SimpleMenu({ getPositionerProps }: InjectedRemirrorProps) {
  return <Menu {...getPositionerProps()} />;
}

export const WrappedSimpleMenu = withRemirror(SimpleMenu);
```
