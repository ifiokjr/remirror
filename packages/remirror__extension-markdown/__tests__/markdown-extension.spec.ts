import { extensionValidityTest, renderEditor } from 'jest-remirror';
import { BoldExtension, HeadingExtension, ItalicExtension } from 'remirror/extensions';

import { MarkdownExtension } from '../';

extensionValidityTest(MarkdownExtension);

describe('commands.insertMarkdown', () => {
  it('can insert blocks', () => {
    const editor = renderEditor([
      new HeadingExtension(),
      new BoldExtension(),
      new ItalicExtension(),
      new MarkdownExtension(),
    ]);
    const { doc, p } = editor.nodes;
    const { bold, italic } = editor.marks;
    const { heading } = editor.attributeNodes;
    const h1 = heading({ level: 1 });

    editor.add(doc(p('Content<cursor>')));

    editor.chain.insertMarkdown('# This is a heading').selectText('end').run();
    expect(editor.state.doc).toEqualProsemirrorNode(doc(p('Content'), h1('This is a heading')));

    editor.commands.insertMarkdown('A paragraph _with_ **formatting**');
    expect(editor.state.doc).toEqualProsemirrorNode(
      doc(
        p('Content'),
        h1('This is a heading'),
        p('A paragraph ', italic('with'), ' ', bold('formatting')),
      ),
    );
  });

  it('can replace blocks', () => {
    const editor = renderEditor([new HeadingExtension(), new MarkdownExtension()]);
    const { doc, p } = editor.nodes;
    const { heading } = editor.attributeNodes;
    const h1 = heading({ level: 1 });

    editor.add(doc(p('<cursor>')));

    editor.chain.insertMarkdown('# This is a heading').selectText('end').run();
    expect(editor.state.doc).toEqualProsemirrorNode(doc(h1('This is a heading')));
  });

  it('can insert multiple blocks', () => {
    const editor = renderEditor([new HeadingExtension(), new MarkdownExtension()]);
    const { doc, p } = editor.nodes;
    const { heading } = editor.attributeNodes;
    const h1 = heading({ level: 1 });

    editor.add(doc(p('<cursor>')));

    editor.chain.insertMarkdown('# Heading\nAnd content').run();
    expect(editor.state.doc).toEqualProsemirrorNode(doc(h1('Heading'), p('And content')));
  });

  it('can insert marks', () => {
    const editor = renderEditor([new BoldExtension(), new MarkdownExtension()]);
    const { doc, p } = editor.nodes;
    const { bold } = editor.marks;

    editor.add(doc(p('Content <cursor>')));

    editor.chain.insertMarkdown('**is bold.**').selectText('end').run();
    expect(editor.state.doc).toEqualProsemirrorNode(doc(p('Content ', bold('is bold.'))));
    expect(editor.dom).toMatchSnapshot();
  });

  it('does not replace marks', () => {
    const editor = renderEditor([new BoldExtension(), new MarkdownExtension()]);
    const { doc, p } = editor.nodes;
    const { bold } = editor.marks;

    editor.add(doc(p('<cursor>')));

    editor.chain.insertMarkdown('**is bold.**').selectText('end').run();
    expect(editor.state.doc).toEqualProsemirrorNode(doc(p(bold('is bold.'))));
    expect(editor.dom).toMatchSnapshot();
  });

  it('can insert marks as blocks', () => {
    const editor = renderEditor([new BoldExtension(), new MarkdownExtension()]);
    const { doc, p } = editor.nodes;
    const { bold } = editor.marks;

    editor.add(doc(p('Content <cursor>')));

    editor.chain
      .insertMarkdown('**is bold.**', { alwaysWrapInBlock: true })
      .selectText('end')
      .run();
    expect(editor.state.doc).toEqualProsemirrorNode(doc(p('Content '), p(bold('is bold.'))));
    expect(editor.dom).toMatchSnapshot();
  });
});
