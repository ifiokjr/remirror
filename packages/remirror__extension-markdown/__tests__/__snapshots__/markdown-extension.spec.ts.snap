// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`commands.insertMarkdown can insert marks 1`] = `
<div
  aria-label=""
  aria-multiline="true"
  class="ProseMirror remirror-editor"
  contenteditable="true"
  role="textbox"
>
  <p>
    Content 
    <strong>
      is bold.
    </strong>
  </p>
</div>
`;

exports[`commands.insertMarkdown can insert marks as blocks 1`] = `
<div
  aria-label=""
  aria-multiline="true"
  class="ProseMirror remirror-editor"
  contenteditable="true"
  role="textbox"
>
  <p>
    Content 
  </p>
  <p>
    <strong>
      is bold.
    </strong>
  </p>
</div>
`;

exports[`commands.insertMarkdown does not replace marks 1`] = `
<div
  aria-label=""
  aria-multiline="true"
  class="ProseMirror remirror-editor"
  contenteditable="true"
  role="textbox"
>
  <p>
    <strong>
      is bold.
    </strong>
  </p>
</div>
`;
